### DISCUSSION TITLE
Improving Ortho4XP (bug fixes, functionality, etc.)

### DISCUSSION THREAD

**Discussion by shred86** [shred86/Ortho4XP#8] [Ideas]
My intent is to not stray too far from the original Ortho4XP application but I do think there's small improvements we could make that can hopefully get merged into the official version. I'm starting this discussion to have a place to discuss issues or ideas that we can improve upon.

**Comment by shred86** [comment]
I think working with configuration files can be confusing and is an area that could be improved upon. A couple of examples:

1. When you select a tile, the configuration for that tile (if it exists) does not load automatically. I think it's more intuitive if it loaded by default.
2. Global configuration doesn't automatically get created when opening the app for the first time. If one isn't found, we should just set defaults and create the configuration file immediately. Otherwise, although it's a niche case, you can find yourself doing some work, try to reload the global configuration defaults but they don't load because the file doesn't exist.
3. It's not easy to identify if you're looking at the global configuration or tile configuration in the Ortho4XP config window. This might require a bigger change but I think the two should really be separate.
4. I'm not sure why the `default_website` (Imagery setting) and `default_zl` (Zoomlevel) are not written to global config but instead, to `.last_gui_params.txt` which is only saved if you use the application's close button (power button). I'm sure there's a reason, but I just can't figure it out yet.

**Comment by w8sl** [comment]
I think Artesim, who created progressive_zl, tried to solve some of these problems (1) It reads zone list from .dsf and displays all info with colors with one button press!
There is no chance progressive_zl feature gets merged to the official master even if it is technically possible, as I have proved on my fork, but it may continue as a separate branch as it is now.

Nr 2 is a problem indeed - especially with progressive_zl, as it causes error when opening the preview window. I have kind of solved it, by placing .cfg file in the repo. From what I see it is not possible to manipulate with configuration file without opening the configuration window.

**Comment by crisk73** [comment]
Hi @shred86, I don't know if this is the right place to ask, but since I noticed that Viewfinderpanoramas' 1 arcsec. DEMs are broken for many areas of the Italian/Austrian Alps, would it be possible to use the more precise and correct LiDAR DEMs from https://sonny.4lima.de/? I mean only for the Alpine tiles in France, Switzerland, Austria and Italy.
As a proof you can try to generate the tile +46+012 with the default mesh and see what the Tre Cime di Lavaredo look like. It seems that some dem data are completely missing. Thanks.
